core/unix/signal.c

```diff
@@ -3946,18 +4031,24 @@ transfer_from_sig_handler_to_fcache_return(dcontext_t *dcontext, kernel_ucontext
 
 #if defined(X64) || defined(ARM)
     /* x64 always uses shared gencode */
-    dcontext->local_state->spill_space.IF_X86_ELSE(xax, r0) =
-        sc->IF_X86_ELSE(SC_XAX, SC_R0);
+    /* FIXME i#3544: RFC: Perhaps IF_<arch>_ELSE ladder should be replaced by
```

Comment:
I guess we do have things like SC_SYSNUM_REG and SC_RETURN_REG where it makes sense. Is there one you suggest here?

Reply:
Since return value in case of a function or syscall follows the same ABI (x86 - xax, ARM - R0, RISC-V - A0), I've added a `retval` register alias in mcxtx and spill_space. Please let me know if you prefer the union approach or rather a define based one as I've described in [one of the fix commits](https://github.com/DynamoRIO/dynamorio/pull/5608/commits/51162039cb20613a4baeca89d64af858eb6da63b).

Reply:
Same as the other comment, I believe it'll be better to keep those structs independent of DR's implementation details.

Reply:
Done.